### PR TITLE
Return expected driver.Value in val.Value()

### PR DESCRIPTION
**Pull request by stephenafamo**

### CODE REVIEW

File: null/primitive_test.go [hunk 1]
```diff
@@ -262,7 +262,7 @@ func TestPrimitives(t *testing.T) {
 		}
 		if b, err := val.Value(); err != nil {
 			t.Error(err)
-		} else if b.(uint16) != 5 {
```
Comment (aarondl) [inline null/primitive_test.go:265]: What's the reason behind this change?

File: internal/globaldata/globaldata.go [hunk 2]
```diff
@@ -4,15 +4,13 @@ import (
...
 	"reflect"
-	"time"
 )
 
 var (
-	JSONNull                          = []byte("null")
-	DriverValuerIntf                  = reflect.TypeOf((*driver.Valuer)(nil)).Elem()
-	EncodingTextMarshalerIntf         = reflect.TypeOf((*encoding.TextMarshaler)(nil)).Elem()
-	EncodingTextUnmarshalerIntf       = reflect.TypeOf((*encoding.TextUnmarshaler)(nil)).Elem()
-	TimeTimeType                      = reflect.TypeOf(time.Time{})
```
Comment (aarondl) [inline internal/globaldata/globaldata.go:15]: Didn't realize this wasn't used. Nice.

File: null/primitive_test.go (hunk 1 above)
Comment (stephenafamo) [reply null/primitive_test.go:265]: In the documentation for `driver.Value`, it says that it is either nil, a type handled by the driver, or one of these types

- int64
- float64
- bool
- []byte
- string
- time.Time

While most drivers handle other integer types just fine, I think it is safest to just send `int64`. Same thing with `float64` for all floats.

File: internal/globaldata/globaldata.go (hunk 2 above)
Comment (stephenafamo) [reply internal/globaldata/globaldata.go:15]: `golangci-lint` to the rescue 😁

File: null/primitive_test.go (hunk 1 above)
Comment (stephenafamo) [reply null/primitive_test.go:265]: Did you get a chance to take another look?

**Review by aarondl — APPROVED** [review]
